scripts/stats.py

```diff
@@ -3,83 +3,266 @@
+                print(str(group_stats[suffix]).rjust(suffix_max_len), end="  ")
             print()
             print()
 
-    totals = collections.defaultdict(int)
-    group_stats = collections.defaultdict(int)
-    last_group = None
-    sort_key = lambda path: (path.group, path.name)  # noqa
-    paths = sorted(paths, key=sort_key)
     for (group, name), g in itertools.groupby(paths, key=sort_key):
         if group != last_group:
             print_group_stats()
             print()
-            print(group.upper().ljust(name_max_len), end='  ')
+            print(group.upper().ljust(name_max_len), end="  ")
             for suffix in suffixes:
-                print(suffix.rjust(suffix_max_len), end='  ')
+                print(suffix.rjust(suffix_max_len), end="  ")
             print()
-            print('-' * name_max_len, end='- ')
+            print("-" * name_max_len, end="- ")
             for suffix in suffixes:
-                print('-' * (suffix_max_len + 1), end=' ')
+                print("-" * (suffix_max_len + 1), end=" ")
             print()
             last_group = group
             group_stats = collections.defaultdict(int)
 
-        print(name.ljust(name_max_len), end='  ')
-        alg_suffixes = {p.suffix for p in g}
+        print(name.ljust(name_max_len), end="  ")
+        alg_suffixes = {extension_map[p.suffix] for p in g}
         for suffix in suffixes:
             if suffix in alg_suffixes:
                 totals[suffix] += 1
                 group_stats[suffix] += 1
-                value = 'x'
+                value = "x"
             else:
-                value = '.'
-            print(value.rjust(suffix_max_len), end='  ')
+                value = "."
+            print(value.rjust(suffix_max_len), end="  ")
         print()
 
     print_group_stats()
 
     print()
-    print('=' * name_max_len, end='= ')
+    print("=" * name_max_len, end="= ")
+    for suffix in suffixes:
+        print("=" * (suffix_max_len + 1), end=" ")
+    print()
+    print("TOTALS:".rjust(name_max_len), end="  ")
+    for suffix in suffixes:
+        print(str(totals[suffix]).rjust(suffix_max_len), end="  ")
+    print()
+
+
+def generate_markdown():
+    global paths, suffixes, name_max_len, suffix_max_len, totals, group_stats, last_group, sort_key
+
+    print(
+        "This is the progress list of [Cosmos](https://github.com/OpenGenus/cosmos/) | Updated on: {}"
+        .format(datetime.today().strftime("%Y-%m-%d")))
+    print()
+
+    def print_group_stats():
+        if group_stats:
+            print("| |", end="")
+            for suffix in suffixes:
+                print(
+                    " {} |".format(
+                        str(group_stats[suffix]).rjust(suffix_max_len)),
+                    end="",
+                )
+            print()
+            print()
+
+    for (group, name), g in itertools.groupby(paths, key=sort_key):
```

Comment:
No need for parens here.

Reply:
@arnavb  i dont understand what change you're asking for. The itertools.groupby returns only two values. It's an error when i tried to remove parens and tried to unpack to 3 diff variables when only two values are there to be unpacked.

Kindly clear your review about which parens you're talking about.

Reply:
Ah, no that was my mistake. I missed the g variable when quickly scrolling through.